hypervisor/virt/esx/powercli.py

```diff
@@ -6,54 +6,139 @@
+        :param admin_user: the user for the vcenter server
+        :param admin_passwd: the password for the vcenter server
+        :param ssh_user: the ssh user for the vcenter server
+        :param ssh_passwd: the ssh password for the center server
+        """
+        self.server = server
         self.admin_user = admin_user
-        self.admin_passwd = admin_passwd 
-        self.ssh_user = ssh_user 
-        self.ssh_passwd = ssh_passwd 
-        self.cert = ("powershell Connect-VIServer -Server {0} -Protocol https -User {1} -Password {2};".format(
-            self.server,
-            self.admin_user,
-            self.admin_passwd)
+        self.admin_passwd = admin_passwd
+        self.ssh_user = ssh_user
+        self.ssh_passwd = ssh_passwd
+        self.cert = (
+            f'powershell Connect-VIServer '
+            f'-Server {self.server} '
+            f'-Protocol https '
+            f'-User {self.admin_user} '
+            f'-Password {self.admin_passwd};'
         )
         self.ssh = SSHConnect(self.server, user=self.ssh_user, pwd=self.ssh_passwd)
-        self.json = 'ConvertTo-Json -Depth 1'
 
     def _format(self, ret=0, stdout=None):
+        """
+        Convert the json string to python list data
+        :param ret: return code
+        :param stdout: output for the execute command
+        :return: the list after json.loads
+        """
         stdout = re.findall(r'[[][\W\w]+[]]', stdout)[0]
-        if ret == 0 and stdout is not None: 
+        if ret == 0 and stdout is not None:
             return json.loads(stdout)
 
     def info(self):
-        mapping = {}
-        ret, output = self.ssh.runcmd("{0} Get-VMHost | {1}".format(self.cert, self.json))
-        logger.info(output)
+        """
+        Get the VMhost info
+        :return: the VMHosts info
+        """
+        cmd = f'{self.cert} ConvertTo-Json @(Get-VMHost | Select * -ExcludeProperty ExtensionData)'
+        ret, output = self.ssh.runcmd(cmd)
         output = self._format(ret, output)
         for host in output:
-            logger.info(host['Name'])
+            logger.info(f"Get ESXi Host: {host['Name']}")
+        return output
 
-    def host_add():
+    def guest_search(self, guest_name):
+        """
+        Search the specific guest, return the expected attributes
+        :param guest_name: name for the specific guest
+        :return: guest attributes, exclude guest_name, guest_ip, guest_uuid ...
+        """
+        cmd = f'{self.cert} ConvertTo-Json @(' \
+            f'Get-VM {guest_name} | Select * -ExcludeProperty ExtensionData)'
+        ret, output = self.ssh.runcmd(cmd)
+        output = self._format(ret, output)[0]
+        host_name = output["VMHost"]["Name"]
+        guest_msgs = {
+            'guest_name': output['Name'],
+            'guest_ip': output["Guest"]["IPAddress"].split()[0],
+            'guest_uuid': self.guest_uuid(guest_name),
+            'guest_state': output["Guest"]["State"],
```

Comment:
As said, we may define a uniform guest state which matched with the mapping info in rhsm.log.

Reply:
Will try some cases to see if the guest state meets our requirements

Reply:
Hey Eko, I prefer to do it with another PR as I still need some other functions to add for now and I still need some time...
what is your thought? Merge this PR if you are OK so that we can move on :)

Reply:
Cool, I will merge it.